horace_core/sqw/@sqw/arrayfun.m

```diff
@@ -1,30 +1,46 @@
 function out = arrayfun(fun,array,varargin)
 % implements matlab's arrayfun for array of sqw objects
 %
-% on 10/06/2017 its a draft inclmplete version missing number of 
+% on 10/06/2017 its a draft inclmplete version missing number of
 % features as a new style classes should have Matlab native arrayfun
 % function availible.
 
+% It does not process multiple fun arguments and fails strangely on finding
+% one
+
 if ~isa(fun,'function_handle')
-    error('SQW:invalid_argument','first argument of arrayfun should be a function handle');
+    error('HORACE:sqw:invalid_argument',...
+        'first argument of arrayfun should be a function handle');
 end
+%
+pa = inputParser();
+pa.KeepUnmatched = true;
+addParameter(pa,'UniformOutput',true,...
+    @(x)(assert(isscalar(x)&&(isnumeric(x)||islogical(x)),...
+    'The UnitormOutput value should be transferrable to logical')));
+
+parse(pa,varargin{:});
+uniformOutput= pa.Results.UniformOutput;
 
-if nargin == 2
-    if nargout>0
+
+if nargout>0
+    if uniformOutput
         tout = fun(array(1));
         out = repmat(tout,size(array));
-        
         for i=2:numel(array)
             out(i) = fun(array(i));
         end
     else
+        out = cell(size(array));
         for i=1:numel(array)
```

Comment:
Could this not be
out = cellfun(array, fun)

Reply:
no -- would not work. Cellfun will not accept array as input. I was not working on this function carefully -- its days are counted until Cris merges new sqw